### PR TITLE
Split implementation out of midi_alsa header

### PR DESCRIPTION
**Pull request by dreamer**
This is preparation for bugfixes in subsequent PRs.

### CODE REVIEW

File: src/midi/midi_alsa.h [hunk 1]
```diff
@@ -29,205 +29,32 @@
 #define ALSA_PCM_OLD_HW_PARAMS_API
 #define ALSA_PCM_OLD_SW_PARAMS_API
 #include <alsa/asoundlib.h>
```
Comment (kcgen) [inline src/midi/midi_alsa.h:31]: Suggest adding a TODO here to use modern 1.0.x ALSA API.

These OLD flags turn on ALSA 0.9.x API (circa 2002), which according to this recent Timidity++ issue, cause the ALSA headers to expose some symbols that cause ambiguous linking, which is a problem for  LTO builds: https://bugs.gentoo.org/763612

File: src/midi/midi_alsa.cpp [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+MidiHandler_alsa::MidiHandler_alsa()
+        : MidiHandler(),
+          ev{},
+          seq_handle(nullptr),
+          seq_client(0),
+          seq_port(0),
+          my_client(0),
+          my_port(0)
+{}
+
```
Comment (kcgen) [inline src/midi/midi_alsa.cpp:92]: Constructor be eliminated by moving these to default member initialization values in the class definition.

(I know you've recommended we use default member initialization before; and it makes even more sense when there are multiple constructors.  But in this case, what's the logic that causes you to prefer one over the other? Just for my own understanding :-) Thank you @dreamer!)

File: src/midi/midi_alsa.cpp [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+		long theBend = ((long)msg[1] + (long)(msg[2] << 7)) - 0x2000;
+		snd_seq_ev_set_pitchbend(&ev, chanID, theBend);
+		send_event(1);
+		break;
+	}
+	default:
+		// Maybe filter out FC as it leads for at least one user to
+		// crash, but the entire midi stream has not yet been checked.
+		LOG(LOG_MISC, LOG_WARN)("ALSA: Unknown Command: %02X %02X %02X",
+		                        msg[0], msg[1], msg[2]);
```
Comment (kcgen) [inline src/midi/midi_alsa.cpp:142]: (I know the goal of this PR isn't to touch the code..)
`msg[0]` in this LOG should probably be `msg[0] & 0xF0`, as that's the exact command that wasn't handled.

**Review by kcgen — APPROVED** [review]
Looks great @dreamer!
Just a couple minor suggestions.

File: src/midi/midi_alsa.cpp (hunk 2 above)
Comment (dreamer) [reply src/midi/midi_alsa.cpp]: Nothing special behind it - I just moved the function as is, reformatted and did some light edits.

I moved initialization to the header. C-tor is trivial now, but it needs to be defined still (also in header).

File: src/midi/midi_alsa.cpp (hunk 3 above)
Comment (dreamer) [reply src/midi/midi_alsa.cpp:132]: Hmm, I'm not sure - this log is shown in the debugger, so maybe it makes sense to show full byte. Leaving it as it is for now.

**Pull request merged**